src/Core/Content/Product/SalesChannel/Listing/ProductListingLoader.php

```diff
@@ -75,7 +75,7 @@ public function load(Criteria $origin, SalesChannelContext $context): EntitySear
             list($variantIds, $mapping) = $this->resolvePreviews($ids->getIds(), $context);
         }
 
-        $read = $criteria->cloneForRead($variantIds);
+        $read = $criteria->cloneForRead(empty($variantIds) ? null : $variantIds);
```

Comment:
This case should not be necessary. The empty check happens on line 60. The mapping should not change the count.

Reply:
Removed!